OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/ReachableMethodAnalysis.scala

```diff
@@ -65,47 +55,41 @@ trait ReachableMethodAnalysis extends FPCFAnalysis with TypeConsumerAnalysis {
             return processMethodWithoutBody(callersEOptP);
 
         val tacEP = propertyStore(method, TACAI.key)
-
         if (tacEP.hasUBP && tacEP.ub.tac.isDefined) {
-            processMethod(callersEOptP, null, tacEP.asEPS)
+            processMethod(callersEOptP, NoCallers, Some(tacEP.asEPS))
         } else {
             InterimPartialResult(Set(tacEP), continuationForTAC(declaredMethod))
         }
     }
 
-    protected val processesMethodsWithoutBody = false
-
     protected def processMethodWithoutBody(
         eOptP: EOptionP[DeclaredMethod, Callers]
-    ): PropertyComputationResult = {
-        if (processesMethodsWithoutBody) {
-            processMethod(eOptP, null, null)
-        } else
-            NoResult
-    }
+    ): PropertyComputationResult = processMethod(eOptP, NoCallers, None)
 
     private[this] def processMethod(
-        eOptP: EOptionP[DeclaredMethod, Callers],
-        seen:  Callers,
-        tacEP: EPS[Method, TACAI]
+        eOptP:      EOptionP[DeclaredMethod, Callers],
+        oldCallers: Callers,
+        tacEPOpt:   Option[EPS[Method, TACAI]]
```

Comment:
An EPS wrapped into an Option looks wrong. EPS is already a subtype of EOptionP which captures the possibility of no value being present.

Reply:
It kind of depends on whether you understand `EPK` as an interim / refinable state (which it says it is since the class expresses it is not final) that *may*  produce a value if awaited via a continuation OR you understand it as a simple "there is no property here and there will never be one".

Since my understanding of the class is the former, I think an `Option[EPS]` is more reasonable than an `EOptionP` here.

Reply:
Well, `EPK` is also used for queries, just representing a pair of entity and propertykey as the name suggest.

Reply:
I tried to use an `EPK`, but there is a problem: Since the `processMethodWithoutBody` is also invoked when no single defined method exists, there may be no `Method` to derive from the given `DeclaredMethod` with which you would create an `EPK[Method, TACAI]`.

So we probably have to go back to using an `Option`, be it `Option[EPS]` or `Option[EOptionP]`. Or do you have another idea?